![](./img/FabLab.svg)

# Week 6. Darwin on steroids: Bio design, diversity & selection

## Lecture video
Link to the [video lecture](https://vimeo.com/140941754)

## Assignment
In class we discussed the idea of a “Human Genome Project 2.0”, but instead of reading DNA, writing DNA. For your homework, please answer each of the following questions:

### If humanity were to undertake such a project, what would be the benefits? What types of new science and engineering would be enabled if we had such a synthetic human genome? Please provide specific examples
Since 1972, the way to engineering genome is open. Still in a early stage, but the walk has started.
<https://www.ncbi.nlm.nih.gov/pubmed/4571075>

Obviously, the first idea that  comes to my mind is to avoid all kind of genetic diseases.  FINDbase was founded in 2006 to be a relational database for these frequencies of causative genetic variations of inherited genetic disorders.
<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3013745/>
<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3964978/>

Avoid getting old, become stronger, increase defenses, fertility, and so on.

The new science probably will be about study the consequences of every change that could be made on the human genome. For instance, if we avoid getting old and at the same time we increase fertility, it would be soon a population problem.

### Conversely, why might we not want to proceed with such an endeavor? What are the risks?
Moral is changing constantly, so what it was good 300 years ago, now is considered bad and vice versa. So besides moral issues, the balance in environment would change with our genome manipulations. Probably the changes will come in many ways that we can not imagine.

###  Map out a technical strategy for synthesizing a human genome. What technologies would be required? What are existing tools we could leverage? For certain tools that do not exist, what should their capabilities be?
I think, the best way to show the needs is with real examples from very importants scientific magazines:
<http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3712628/>

Here, we engineer the protein and RNA components of this bacterial type II CRISPR system in human cells. We began by synthesizing a human codon–optimized version of the Cas9 protein bearing a C-terminal SV40 nuclear localization signal and cloning it into a mammalian expression system (Fig. 1A and fig. S1A). To direct Cas9 to cleave sequences of interest, we expressed crRNA-tracrRNA fusion transcripts, hereafter referred to as guide RNAs (gRNAs), from the human U6 polymerase III promoter. Directly transcribing gRNAs allowed us to avoid reconstituting the RNA-processing machinery used by bacterial CRISPR systems (Fig. 1A and fig. S1B) (4, 7-9). Constrained only by U6 transcription initiating with G and the requirement for the PAM (protospacer-adjacent motif) sequence -NGG following the 20–base pair (bp) crRNA target, our highly versatile approach can, in principle, target any genomic site of the form GN20GG (fig. S1C; see supplementary text S1 for a detailed discussion).

We can find more clues on princeton university web page:
<http://www.princeton.edu/genomics/botstein/protocols/>

Mating, Sporulation, Tetrad dissection, Colony PCR, DNA preparation (decant, resuspend in water, spin in centrifuge, air dry, and so on.), ...

## What I learned
In a short period of time we are going to be able to do all kind of modifications to species at genomic level, having access to door that so far has only be opened for God and natural evolution. The fascination that this produces is quickly eclipsed by the fear of the question what will be the next atrocity human being will produce with this knowledge.

## Assignment review
On Wednesdays we always have a review session of last week's assignment. Here is the link to this week [assignments review](https://vimeo.com/141797544).

---

## [<<](./w05.html)  [  home  ](./index.html)  [  >>](./w07.html)
